integr8ly/integreatly-operator
# feat: alert for failed deletion of redis and postgres resources

# Description


Adds new alerts specifically for the failed deletion of Postgres and Redis CRs. This realistically should only happen on teardown of RHMI.

Is a follow on from this PR (still in review) https://github.com/integr8ly/cloud-resource-operator/pull/159

Still todo:

* Create new SOPs for handling failed deletion of Redis and Postgres instances. cc @aidenkeating 

## Verification Steps

Setup:
* Provision the integreatly operator, ensuring that v0.18.0 or the master branch of the cloud-resources-operator is used. This is needed because the cro now exposes some new metrics when AWS resources are being deleted.
* ensure that `useClusterStorage` is `'false'` in the `rhmi` CR.

Verify the alerts were created:

* run the command `oc get prometheusrule -n redhat-rhmi-operator | grep resource-deletion-status-phase-failed-rule`. You should see there are a bunch of prometheus rules for the failed deletion of postgres instances for each product and then the two redis instances that 3scale uses.

Trigger a failed deletion and observe the alerts:

* in the `redhat-rhmi-operator` namespace, find the aws strategies configmap and under the `createStrategy` JSON object for both `'postgres'` and `'redis'` there should be `region` set to `""`. Change the region for both postgres and redis to some dummy value like `"abc123"`. This will cause all AWS operations performed by CRO to fail.
* In the `redhat-rhmi-operator` namespace `oc get postgres` should give you a list of all the various postgres instances being used by the RHMI products. Delete one of those postgres CRs.
* In the alerts tab in the middleware monitoring prometheus you should be able to find the PostgresDeletionFailed alert for the specific product whose postgres instance you tried to delete. It should show up as pending and eventually after about 5 minutes it should start firing. You can also search in the metrics tab for `cro_postgres_deletion_timestamp` and you should get back some metrics.
* repeat the same process for one of the threescale redis instances and observe that there are some `cro_redis_deletion_timestamp` metrics and also observe the alerts moving from a pending state, eventually into a firing state.

## Type of change



- [ x ] New feature (non-breaking change which adds functionality)

## Checklist
